source/kaleidic/lubeck2.d

```diff
@@ -2044,6 +2065,83 @@ unittest
     assert(equal!approxEqual(res.eigenvalues, eigenvalues));
 }
 
+/++
+Computes Moore-Penrose pseudoinverse of matrix.
+
+Params:
+    matrix = Input `M x N` matrix.
+    tolerance = The computation is based on SVD and any singular values less than tolerance are treated as zero.
+Returns: Moore-Penrose pseudoinverse matrix
++/
+@safe pure @nogc nothrow
+Slice!(RCI!T, 2)
+    pinv(T, SliceKind kind)(Slice!(const(T)*, 2, kind) matrix, double tolerance = double.nan)
+{
+    import mir.algorithm.iteration: find, each;
+    import std.math: nextUp;
+
+    size_t info;
+    bool lowApiExecuted;
+
+    auto svd = svdImpl(matrix, info, lowApiExecuted);
+
+    if(lowApiExecuted)
+    {
+        enum msg ="pinv: pinv was not successful due to a convergence issue during SVD calculation.";
+        assert(!info,  msg);
```

Comment:
Do you need convergence validation for pinv function?
If so, then please do the same: two implementations of `pinv`, where one of them is has the info out argument.